### ISSUE TITLE
Sign-off page: form data is persisted in a database model.

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-reporting#451]
#### Description:

Right now, there is no database record of the checkboxes a user clicked, the name and the date filled on the sign-off form.
We need a model to persist that data before marking the report as submitted

#### Acceptance Criteria:

Given
When
Then

#### Development Checklist:

- [ ] create model/migration
- [ ] have sign-off page submit save data to new table
- [ ] have sign-off page load from new table on load
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking

Blocking ticket [603](https://github.com/bcgov/cas-reporting/issues/603)
#### Notes:

-

**Comment by patriciarussellCAS** [comment]
Hi @pbastia - can we discuss this ticket next week? It seems like a high priority if the system is not properly capturing the name and date when the report is submitted. But I may not be understanding the ticket.

**Comment by shon-button** [comment]
Blocking ticket [603](https://github.com/bcgov/cas-reporting/issues/603)

**Comment by shon-button** [comment]
SFO:

![Image](https://github.com/user-attachments/assets/d7c191c8-b473-44b1-a5c4-456f24d1dec9)

LFO:

![Image](https://github.com/user-attachments/assets/8a787b3f-ec4e-42d7-ac1e-17a75927f5a0)

SIMPLE:

![Image](https://github.com/user-attachments/assets/2d62ca2c-0d33-429a-80de-fb5c5b5c0afe)

SUP REPORTING:

![Image](https://github.com/user-attachments/assets/d7104c9a-6ac4-46bf-af26-56fcac4a9af8)

SUP EIO:

![Image](https://github.com/user-attachments/assets/d827b3ee-514a-4bb8-95b2-71527ce7f9ab)

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EeQomWnT26ZHqVaqOPDvFycBUa_g8RIkxsawIxHohaqyjg?e=PrlQib).

I can verify that:
- [x] When verification is required for a report, expected errors are encountered in the Sign-off page when required data are not provided (Data in Verification page and Verification Statement)
- [x] When verification is required for a report, no errors are encountered in the Sign-off page when required data are provided (Data in Verification page and Verification Statement)
- [x] When verification is NOT required for a report, no errors are encountered when submitting report without Verification Statement 
- [x] When report is submitted, the data is visible in the `report_sign_off` table
- [x] When report is submitted, the status correctly changes in the reporting grid to Submitted
- [x] When supplementary report is created, the Sign-off page is not filled out 
- [x] When all checkboxes in the Sign-off page are not ticked, the `Submit Report` button is disabled
- [ ] When all checkboxes in the Sign-off page are ticket, but signature is not provided, the `Submit Report` button is disabled - currently the `Submit Report` button gets enabled when all checkboxes are ticked even though no signature is provided, once user submits, an error banner is displayed without any text, suggesting to not enable the button if all required fields are not provided yet #673

Moving to Pending PO for review of test and notes. Thank you!

**Comment by patriciarussellCAS** [comment]
Thanks @fviduya and good catch with the submit page. Moving to accepted.
